Project.toml

```
@@ -0,0 +1,28 @@
+name = "ImageCore"
+uuid = "a09fc81d-aa75-5fe9-8630-4744c3626534"
+version = "0.8.0"
+
+[deps]
+ColorTypes = "3da002f7-5984-5a60-b8a6-cbb66c0b333f"
+Colors = "5ae59095-9a9b-59fe-a467-6f913c188581"
+FFTW = "7a1cc6ca-52ef-59f5-83cd-3a7055c09341"
+FixedPointNumbers = "53c48c17-4a7d-5ca2-90c5-79b7896eea93"
+Graphics = "a2bd30eb-e257-5431-a919-1863eab51364"
+MappedArrays = "dbb5928d-eab1-5f90-85c2-b9b0edb7c900"
+OffsetArrays = "6fe1bfb0-de20-5000-8ca7-80f57d26f881"
+PaddedViews = "5432bcbf-9aad-5242-b902-cca2824c8663"
+
+[compat]
+julia = ">= 0.7"
```

```suggestion
julia = ">= 1.0"
```
if we're going to drop testing on 0.7.

---

🤔 My thought is: compatibility doesn't equal to supportability. Not supporting julia 0.7 with CI test doesn't mean the codes can't be run on julia 0.7 version.

---

The problem is this: what if we introduce a commit that breaks 0.7 but we don't realize it? Now someone running julia 0.7 tries out their code, does `]add ImageCore`, and it adds the broken version? Then they have to manually figure out the last version we supported and pin it there. In contrast, if we drop 0.7 compatibility, then at least we guarantee a working version of this package on 0.7.

The other approach is to keep testing 0.7.

---

I've taken this problem into consideration. Since 0.7 is not officially supported, users shall expect bugs if they're using a newer version of `ImageCore`. In this case, a manual test can help figure out the problem
```julia
]activate .
]instantiate
]test
```

But I think you're right, at least not all users can easily understand all kinds of error messages. Throwing a version conflict error makes their life easier.

I'll change this tomorrow as well as other PRs(including merged ones), it's quite late here.

---

In general it wouldn't even be a version conflict; Pkg should give them a mutually-compatible set of packages. They'd basically be stuck on the 0.7 series of ImageCore.

---

I've dropped compatibility of all packages to julia 0.7 except the TestImages since it's quite trivial. Packages that still has 0.7 under CI test are not touched, e.g., ImageView